lib/core/versions/latest/DocumentComposer.ts

```diff
@@ -197,25 +186,61 @@ export default class DocumentComposer {
   }
 
   /**
-   * Validates update patch for either adding or removing service endpoints.
+   * validate update patch for removing service endpoints
    */
-  private static validateServiceEndpointsPatch (patch: any) {
-    const patchProperties = Object.keys(patch);
-    if (patchProperties.length !== 3) {
-      throw new SidetreeError(ErrorCode.DocumentComposerPatchMissingOrUnknownProperty);
+  private static validateRemoveServiceEndpointsPatch (patch: any) {
+    if (!Array.isArray(patch.serviceEndpointIds)) {
+      throw new SidetreeError(ErrorCode.DocumentComposerPatchServiceEndpointIdsNotArray);
     }
 
-    if (patch.serviceType !== 'IdentityHub') {
-      throw new SidetreeError(ErrorCode.DocumentComposerPatchServiceTypeMissingOrUnknown);
+    for (const id of patch.serviceEndpointIds) {
+      if (typeof id !== 'string') {
+        throw new SidetreeError(ErrorCode.DocumentComposerPatchServiceEndpointIdsIdNotString);
+      }
+      if (id.length > 20) {
+        throw new SidetreeError(ErrorCode.DocumentComposerPatchServiceEndpointIdsIdTooLong);
+      }
     }
+  }
 
+  /**
+   * Validates update patch for adding service endpoints.
+   */
+  private static validateAddServiceEndpointsPatch (patch: any) {
```

Comment:
Related to the comment above, the array check is common and can move into `validateServiceEndpoints`. Also probably need to add check for patch property length like `validateAddPublicKeysPatch()`, so that patch with unknown properties won't be considered valid.